openlibrary/catalog/get_ia.py

```diff
@@ -79,14 +82,19 @@ def get_marc_record_from_ia(identifier):
     # Try marc.xml first
     if marc_xml_filename in filenames:
         data = urlopen_keep_trying(item_base + marc_xml_filename).read()
-        if data[:10].find('<?xml') != -1:
+        try:
             root = etree.fromstring(data)
             return MarcXml(root)
+        except Exception as e:
```

Comment:
I agree with @cdrini that `Exception` is way too broad. The next time someone introduces a syntax or logic error in this block it'll be totally masked.

@hornc if there are a range of exceptions that can be thrown, can they be listed (or their superclass)?